### PR TITLE
Refactor parsing for greater flexibility in parsing JSON, etc.

### PR DESCRIPTION
**Pull request by akarve**
Invert control (focus on file extensions / transforms) and make pandas secondary so that we can use python native parsers, etc. for new file types

### CODE REVIEW

File: quilt/tools/build.py [hunk 1]
```diff
@@ -96,22 +100,23 @@ def _file_to_spark_data_frame(ext, path, target, user_kwargs):
...
-        raise BuildException('Unsupported target platform: %s' % target)
-    logic = platform.get(ext)
-    if logic is None:
-        raise BuildException(
-            "Unsupported transform: %s. Try setting a 'transform' key." % ext)
-    fname = logic['attr']
+    logic = PARSERS.get(ext)
+    module_name = logic['module']
+    the_module = None
+    if module_name in sys.modules:
```
Comment (dimaryaz) [inline quilt/tools/build.py]: I'm pretty sure it's not necessary to check if the module is already imported

File: quilt/tools/build.py [hunk 2]
```diff
@@ -96,22 +100,23 @@ def _file_to_spark_data_frame(ext, path, target, user_kwargs):
...
+        the_module = sys.modules[module_name]
+    else:
+        the_module = importlib.import_module(module_name)
```
Comment (dimaryaz) [inline quilt/tools/build.py]: I don't really like the idea of importing arbitrary modules: e.g., you might run arbitrary code just by building a package. Maybe whitelist module names?

Comment (akarve) [reply quilt/tools/build.py]: OK but this is in our code in const.py and if they can modify PARSERS they can modify the whitelist. I can see your objection if the modules are specified in build.yml but that's not the case right?

Comment (dimaryaz) [reply quilt/tools/build.py]: Oh whoops, I misunderstood - thought it came from build.yml. That's fine then.

**Pull request merged**